apps/web/src/components/tx/confirmation-views/useTxPreview.ts

```diff
@@ -2,6 +2,17 @@ import { Operation, getTxPreview } from '@safe-global/safe-gateway-typescript-sd
 import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
 import useAsync from '@/hooks/useAsync'
 import useSafeInfo from '@/hooks/useSafeInfo'
+import { sameAddress } from '@/utils/addresses'
+import { Interface } from 'ethers'
+
+export const InvalidPreviewErrorName = 'InvalidPreviewError'
+class InvalidPreviewError extends Error {
+  constructor(message: string) {
+    super(message)
+
+    this.name = InvalidPreviewErrorName
+  }
+}
 
 const useTxPreview = (
```

Comment:
This hook was very simple – just an RTK call. Now it does more. Can we extract the validation part into a separate hook (can be in the same file) and call it from the old hook?